### ISSUE TITLE
links test has unreported failures

### ISSUE THREAD

**Issue by qkoziol** [HDFGroup/hdf5#4672]
**Describe the bug**
When running the 'links' test, it reports failures in the output, but doesn't cause the test to fail.

**Expected behavior**
Not fail

**Platform (please complete the following information)**
Current develop branch on MacOS

**Additional context**

Sample output:

```
============================
Test log for links
VOL connector: pass_through under_vol=1;under_info={under_vol=0;under_info={}}
============================

-Testing with unminimzed dataset object headers-

--Testing with 'old format'--
Testing link creation                                                  PASSED
Testing link queries                                                   PASSED
Testing link queries using deprecated routines                         PASSED
Testing H5Lcreate functions                                            PASSED
Testing new link queries                                               PASSED
Testing long names for objects & links                                 PASSED
Testing too many links                                                 PASSED
Testing H5Lcreate path handling                                        PASSED
Testing link creation property lists                                   PASSED
Testing link creation property lists using deprecated routines         PASSED
Testing H5Lmove                                                        PASSED
Testing H5Lcopy                                                        PASSED
Testing moving and copying links preserves their properties            PASSED
Testing moving and copying links using deprecated routines preserves their properties PASSED
Testing backwards compatibility                                        PASSED
Testing external file cache with external links                        PASSED
Testing external links across multiple files                           PASSED
Testing H5Pset/get_elink_acc_flags()                                   PASSED

---Testing without external file cache---
Testing external link to root                                          PASSED
Testing external link to root using deprecated routines                PASSED
Testing external link to object on path                                PASSED
Testing external link to self                                          PASSED
Testing external links back and forth                                  PASSED
Testing too many external links                                        PASSED
Testing dangling external links                                        PASSED
Testing recursive external links                                       PASSED
Testing query aspects of external link                                 PASSED
Testing query aspects of external link using deprecated routines       PASSED
Testing unlinking external link in compact group                       PASSED
Testing unlinking external link in dense group                         PASSED
Testing move external link                                             PASSED
Testing external link along for the ride                               PASSED
Testing that external files are closed during traversal                PASSED
Testing that external files are closed during traversal                PASSED
Testing endianness of external links                                   PASSED
Testing that external files work with strong file close degree         PASSED
Testing external links via H5Pset_elink_prefix()                       PASSED
Testing external links via main file's absolute path                   PASSED
Testing external links via main file's CWD + relative path             PASSED
Testing external links via current working directory                   PASSED
Testing external links via target's absolute path                      PASSED
Testing external links via main file's CWD                             PASSED
Testing external links via main file's CWD + target's relative path    PASSED
Testing external links via chdir and found in current working directory PASSED
Testing H5Pset/get_elink_fapl() with different physical layouts       *FAILED*
*FAILED*
*FAILED*
*FAILED*
 PASSED
Testing H5Pset/get_elink_fapl() with same physical layout              PASSED
Testing H5Pset/get_fapl()                                              PASSED
Testing inheriting of file locking settings                            PASSED
Testing H5Pset/get_elink_cb()                                         *FAILED*
*FAILED*
*FAILED*
 PASSED
```

**Comment by bmribler** [comment]
At times, H5FDopen() is inside the H5E_PAUSE_ERRORS/H5E_RESUME_ERRORS or H5E_BEGIN_TRY/H5E_END_TRY block, but the private call H5FD_open is not.  Sometime, "try" is passed in as false causing this difference between the API and internal function too.
When the internal function failed, like H5FD_open failed in H5FDopen() as below:
         if (H5FD_open(false, &ret_value, name, flags, fapl_id, maxaddr) < 0)
             HGOTO_ERROR(H5E_VFL, H5E_CANTINIT, NULL, "unable to open file");

\*FAILED\* was printed.  Please advice on how this issue can be resolved.

**Comment by bmribler** [comment]
@qkoziol Hi Quincey, in case you missed my comment above...  Thanks!

**Comment by qkoziol** [comment]
> @qkoziol Hi Quincey, in case you missed my comment above... Thanks!

Ah, sorry, I missed this earlier.  I'll take a look at it and see how to resolve the issue.

**Comment by bmribler** [comment]
[like]  Binh-Minh Ribler reacted to your message:
________________________________

**Comment by qkoziol** [comment]
PR to fix this is up:   https://github.com/HDFGroup/hdf5/pull/4809

**Comment by derobins** [comment]
Fixed in #4809
